src/operations/blas3/gemm_load_store.hpp

```diff
@@ -36,19 +36,19 @@ used no matter what value is passed here.
 * @tparam value_t The type of the matrix data (typically float or double, if
 supported).
 */
-template <size_t vector_size, typename value_t, typename index_t>
```

Comment:
Rewrite the template so that the index_t will be at the beginning and then you can re-use it for the vector_size. This can be on a separate PR maybe as it requires other files calling it to be modified.